src/main/java/frc/robot/subsystems/shooter/ShooterIOTalonFX.java

```diff
@@ -0,0 +1,294 @@
+    this.shootMotorRightRequestedVelocityRPS = rps;
+  }
+
+
+  @Override
+  public void setAngle(double angle) {
+    // Magic Motion
+  }  
+
+
+  private void configShootMotors(int leftMotorID, int rightMotorID){
+      shootMotorLeft = new TalonFX(leftMotorID, RobotConfig.getInstance().getCANBusName());
+      shootMotorRight = new TalonFX(rightMotorID,RobotConfig.getInstance().getCANBusName());
+
+      TalonFXConfiguration shootMotorsConfig = new TalonFXConfiguration();
+      CurrentLimitsConfigs shootMotorsCurrentLimits = new CurrentLimitsConfigs();
+
+      shootMotorsCurrentLimits.SupplyCurrentLimit = ShooterConstants.SHOOT_MOTORS_CONTINUOUS_CURRENT_LIMIT;
+      shootMotorsCurrentLimits.SupplyCurrentThreshold = ShooterConstants.SHOOT_MOTORS_PEAK_CURRENT_LIMIT;
+      shootMotorsCurrentLimits.SupplyTimeThreshold = ShooterConstants.SHOOT_MOTORS_PEAK_CURRENT_DURATION;
+      shootMotorsCurrentLimits.SupplyCurrentLimitEnable = true;
+
+      shootMotorsConfig.CurrentLimits = shootMotorsCurrentLimits;
+
+      shootMotorsConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+      shootMotorsConfig.Slot0.kP = shootMotorsKP.get();
+      shootMotorsConfig.Slot0.kI = shootMotorsKI.get();
+      shootMotorsConfig.Slot0.kD = shootMotorsKD.get();
+
+      shootMotorsConfig.MotorOutput.Inverted =
+        ShooterConstants.SHOOT_LEFT_INVERTED // TODO: What Should I do for the Right Shoot Motor?
+          ? InvertedValue.Clockwise_Positive
+          : InvertedValue.CounterClockwise_Positive;
+
+      shootMotorLeft.getConfigurator().apply(shootMotorsConfig);
+      shootMotorRight.getConfigurator().apply(shootMotorsConfig);
+  }
+
+  private void configAngleMotor(int angleMotorID, int angleEncoderID){
+    angleMotor = new TalonFX(angleMotorID, RobotConfig.getInstance().getCANBusName());
+    angleEncoder = new CANcoder(angleEncoderID);
+
+    TalonFXConfiguration angleMotorConfig = new TalonFXConfiguration();
+    CurrentLimitsConfigs angleMotorCurrentLimits = new CurrentLimitsConfigs();    
+    
+    // TODO: how to do want the CAN Configuration to be done?
+
+
+    angleMotorCurrentLimits.SupplyCurrentLimit = ShooterConstants.ANGLE_MOTOR_CONTINUOUS_CURRENT_LIMIT;
+    angleMotorCurrentLimits.SupplyCurrentThreshold = ShooterConstants.ANGLE_MOTOR_PEAK_CURRENT_LIMIT;
+    angleMotorCurrentLimits.SupplyTimeThreshold = ShooterConstants.ANGLE_MOTOR_PEAK_CURRENT_DURATION;
+    angleMotorCurrentLimits.SupplyCurrentLimitEnable = true;
+
+    angleMotorConfig.CurrentLimits = angleMotorCurrentLimits;
+
+    angleMotorConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    angleMotorConfig.Slot0.kP = rotationMotorKP.get();
+    angleMotorConfig.Slot0.kI = rotationMotorKI.get();
+    angleMotorConfig.Slot0.kD = rotationMotorKD.get();
+
+    angleMotorConfig.MotorOutput.Inverted = ShooterConstants.ANGLE_MOTOR_INVERTED
+      ? InvertedValue.Clockwise_Positive
+      : InvertedValue.CounterClockwise_Positive;
+
```

Comment:
We also need to configure fusing the CANcoder with this motor. More details here: https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/device-specific/talonfx/remote-sensors.html#fusedcancoder